libfwupdplugin/fu-udev-device.c

```diff
@@ -479,24 +479,21 @@ gboolean
 fu_udev_device_parse_number(FuUdevDevice *self, GError **error)
 {
 	FuUdevDevicePrivate *priv = GET_PRIVATE(self);
-	g_autoptr(GString) path = g_string_new(fu_udev_device_get_sysfs_path(self));
+	const gchar *sysfs = fu_udev_device_get_sysfs_path(self);
+	g_auto(GStrv) tokens = fu_strsplit(sysfs, strlen(sysfs), "/", -1);
 
-	if (path->len == 0)
-		return TRUE;
-	for (guint i = path->len - 1; i > 0; i--) {
-		if (!g_ascii_isdigit(path->str[i])) {
-			g_string_erase(path, 0, i + 1);
-			break;
-		}
-	}
-	if (path->len > 0) {
-		if (!fu_strtoull(path->str,
+	for (guint i = 0; tokens[i] != NULL; i++) {
+		if (!g_str_has_prefix(tokens[i], "i2c-"))
+			continue;
```

Comment:
I don't think putting i2c here is a good idea. Can we handle this case in the FuI2cDevice file?

Reply:
fixed